test/test_jh_utils.py

```diff
@@ -15,10 +17,19 @@ def initialize_jh_env(server):
 
 
 def test_import_fails_without_initialization():
-    # this fails because proper env variables were not set up
-    with pytest.raises(Exception) as exec_info:
-        from dcicutils import jh_utils  # NOQA
-    assert 'ERROR USING JUPYTERHUB_UTILS!' in str(exec_info.value)
+
+    # Loading this library fails if proper env variables were not set up.
+    # TODO: I'm not sure I think that's a good idea. Functions should fail, imports should not. -kmp 14-Aug-2020
```

Comment:
Imports are functions to me 😄   - IMO imports should fail if on import the module determines it will not work, as is the case with this particular module. I would rather fail on import than fail mysteriously in a function that I shouldn't have been able to call in the first place.

Reply:
I get what you're saying. But then again it makes it hard to unit test. It means that static code analysis tools trip over it, thinking perhaps that the entire file is unreachable. The model I'm suggesting is that there are more reasons to load a library than to execute it.